can/exceptions.py

```diff
@@ -0,0 +1,61 @@
+"""
+There are several specific :class:`Exception` classes to allow user
+code to react to specific scenarios related to CAN busses::
+
+    Exception (Python standard library)
+     +-- ...
+     +-- CanError (python-can)
+         +-- CanBackendError
+         +-- CanInitializationError
+         +-- CanOperationError
+         +-- CanTimeoutError
+
+Keep in mind that some functions and methods may raise different exceptions.
+For example, validating typical arguments and parameters might result in a
+:class:`ValueError`. This should be documented for the function at hand.
+"""
+
+
+class CanError(Exception):
+    """Base class for all CAN related exceptions."""
+
+
+class CanBackEndError(CanError):
+    """Indicates an error related to the backend (e.g. driver/OS/library).
+
+    Example scenarios:
+      - The interface does not exist

```

Comment:
I need some feedback on this: What error should be raised if the interface does not exist (on this platform)? Also a `CanBackEndError`? That's how it is done in the proposal you are reading.

One could also consider `ImportError`, not `NotImplementedError` or a specific sub-class Exceptions that may inherit from any combination of them.

Reply:
How about a specific exception e.g., `CanInterfaceNotImplemented` inheriting from `CanError` and `NotImplementedError`?

Reply:
Sounds good!